public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -530,7 +578,15 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
         hasServiceMapQuery: targets.serviceMap[0].serviceMapQuery ? true : false,
       });
 
-      const { datasourceUid, histogramType } = this.serviceMap;
+      const { datasourceUid } = this.serviceMap;
+
+      let histogramType = undefined;
+      // if the query contains the serviceMapUseNativeHistograms flag,
+      // then use the native histograms
+      if (options.targets[0].serviceMapUseNativeHistograms) {
+        histogramType = 'native';
```

Comment:
We can probably remove this type and just set serviceMapUseNativeHistograms to `true`? I presume this means removing the datasource setting - which would be done in a follow up PR. The setting was [pushed](https://github.com/grafana/grafana/pull/105989) over a month ago, so not sure if it should be officially deprecated cc @ifrost

Reply:
Yes! This work can remove the data source setting which is very new. We have not yet done any work to update this in the Stack State Service and because native histograms have not been rolled out in Tempo yet, it is highly unlikely anyone is using this config option.

There is also the option that we could use the config option as an override. If a user sets it, that would take precedence, so that they could choose to render classic histograms if they wanted. Supporting classic histograms feels weird to me, but supporting legacy Prometheus things is not weird (for example the series endpoint which people have been trying to move away from for years).

Reply:
> This work will allow for removing the data source setting (...) Supporting classic histograms feels weird to me (...)

Would this be a breaking change? Customers who only have classic histograms won't be able to use them anymore without the setting in the date source, would they?

Reply:
Oh, nvm. If they don't have native histograms the flag in the query won't be set because of auto-discovery logic 👍

Reply:
Classic histograms are still supported as a fallback when there are no native histograms returned.